lua/core/clock.lua

```diff
@@ -1,7 +1,13 @@
+--- clock coroutines
+-- @module clock
+
 local clock = {}
 
 clock.threads = {}
 
+--- create a coroutine from the given function and immediately run it. 
+-- @tparam function f
+-- @treturn integer : coroutine ID that can be used to stop it later
```

Comment:
the function parameter could be described better, we expect a certain kind of function here (a task within a script) that will suspend when `clock.sleep` and `clock.sync` are called inside it and wake up again after specified time.